### ISSUE TITLE
Client lockup when moving to new (underground) areas

### ISSUE THREAD

**Issue by TWCCarlson** [ScreteMonge/3D-Weather#32]
Hello,

I've really enjoyed this plugin for a long time and typically always leave it on. While doing clues recently I started to run into client freezes:
![image](https://github.com/ScreteMonge/3D-Weather/assets/72635603/21d3f385-a1e4-46a4-b61c-fe9d1718ce7c)

The game stops rendering and seems un-interactable, even when toggling the GPU plugin.

I found I was able to consistently recreate this by using the minecart from the GE to Keldagrim, but not in RuneLite safe mode. Iterating through my plugin hub plugins, I found 3D Weather to be the consistent problem. Seems to be independent of the GPU plugin being on or off.

It seemed to only happen when transitioning from the overworld to an underground area, as I can cause a freeze going from GE to Keldagrim, or from Edgeville to Edgeville dungeon, but not the other way around. These are my settings for the plugin (I like rain sounds):
![image](https://github.com/ScreteMonge/3D-Weather/assets/72635603/be38722d-42b5-4499-9fa6-7ba75b7d7297)

With "Disable Weather Underground" unticked however, I don't find any issues.

It seems to have to do with the weather type setting. These ones cause the client to freeze:
- Stormy
- Snowy
- Rainy

Interestingly, ashfall doesn't cause any problems despite being another one-tile effect (I did enable it to test).

Happy to provide any more info that could be of help.

**Comment by ScreteMonge** [comment]
Thanks for the heads up, I'll look into it tomorrow

**Comment by ScreteMonge** [comment]
Thanks again for the report as it's very thorough. I've done a bit of testing but couldn't replicate the bug. I haven't seen any other reports like this so far so it might be a niche problem, but of course one that should get sorted out.

Would you mind triggering the crash again and then sending me the most recent client.txt file? You'll usually find the file in some address like this: C:\Users\[Your_Name]\.runelite\logs

**Comment by TWCCarlson** [comment]
I was able to recreate the crash but experienced some different behavior while trying to do so.

I started by using the GE-Keldagrim cart and seemed to have no issue while switching things around:
- Switching weather types between snowy and rainy seemed fine
- Disabling or not disabling weather underground was also fine
- GPU plugin enabled or disabled was fine

I was about to say the issue had resolved itself despite no software updates or anything to my machine, but...

It was only when I walked away a few chunks to the Edgeville dungeon trapdoor that I was able to trigger the crash again. Here's the log for that:
[client.log](https://github.com/ScreteMonge/3D-Weather/files/15082778/client.log)

Same thing occurs with GPU plugin on, as before. I'm a bit confused by the log file as it seems to just tack on new data, and inconsistently append it sometimes. With the GPU on the log file became:
[client.log](https://github.com/ScreteMonge/3D-Weather/files/15082818/client.log)

Which doesn't terminate with the same nullpointerexception. I didn't need to move chunks here, just used the same ladder. Returning the to minecart transition to try again after a few times at the edgeville ladder gave me a consistent crash, and this file:
[client.log](https://github.com/ScreteMonge/3D-Weather/files/15083045/client.log)

Seems weird.

**Comment by TWCCarlson** [comment]
I realized the configure menu has a debug option and triggered another crash with it enabled:
[client.log](https://github.com/ScreteMonge/3D-Weather/files/15083456/client.log)

**Comment by TWCCarlson** [comment]
Also was able to trigger the crash by switching "Disable Weather Underground" to True while underground, not sure why I didn't try that before:
[client.log](https://github.com/ScreteMonge/3D-Weather/files/15083471/client.log)

**Comment by ScreteMonge** [comment]
Very strange indeed. I've replicated all the steps you've given me without success, including doing a ton of area swaps up/down edgeville dungeon and GE-Keldagrim, checking and unchecking the Disable Weather Underground feature.

Looking at your logs, nothing particularly strikes me as out of the ordinary. There are a few NullPointerExceptions relating to the Path Marker plugin you've installed, but I'm not sure if that's directly related to the crash. Note that the logs do append on top of each other - every irl day will produce a new log file, which will keep growing until the next day when a new log file is generated.

I did note that you have a fair few Hub Plugins installed. As a general rule, the less Hub Plugins you have the less likely you're going to run into trouble (including with mine). I tried installing a few of yours to see if it was one of them. I did manage to crash once with 3D Weather, Path Marker, and Random Event Analytics on when descending the Edgeville ladder. I couldn't replicate it, though, no matter how I tried. And just like your log files, nothing clearly broke.

If you'd like to run RL in safemode again, turn on 3D weather and try test with that being the only plugin active, you may do so and resend the new log file if you'd like to get to the bottom of this. As it stands, though, we may just have to wait for someone else to come out with a similar report to see where the pattern is

**Comment by TWCCarlson** [comment]
Very understandable.

To make things even weirder and my-setup-specific, I started testing with music enabled to see if the music continues beyond the freeze. And, well, it appears to be somehow related to the audio:
https://www.youtube.com/watch?v=Rki81_CHIyU

I use an application called VoiceMeeter to manage audio sources and outputs. Clearly some interaction with a RL launch and these settings isn't working correctly as I receive no audio when using VoiceMeeter inputs. This turns into a freeze if I attempt to go underground with some combination of plugins. If I switch the source after the launch there's no issues, including no freezing.

I think that indicates this problem isn't really related to your plugin at all which is good. I'll try more permutations of stuff and see if I can figure out some kind of root cause eventually.

**Comment by ScreteMonge** [comment]
Thanks for figuring it out, definitely brings me some relief that it's not related to the plugin (or at least, not wholly related to it).

**Comment by TWCCarlson** [comment]
Just looking into how RL handles audio and trying to get an overview, I came across this comment in your files; I'd be curious to know what you meant by "massive issues" if you remember:
https://github.com/ScreteMonge/3D-Weather/blob/01ffb81b720aac0d136e1d0e37e121cdb42aee7a/src/main/java/com/weather3d/audio/SoundPlayer.java#L42C9-L42C138

**Comment by TWCCarlson** [comment]
I've learned there's a debug plugin that lets me do a thread dump:
[message.txt](https://github.com/ScreteMonge/3D-Weather/files/15112881/message.txt)

Scanning through it I can see that the other thread states are waiting on something on the object monitor to finish doing it's thing. Specifically, "Thread-32", "Thread-37", "Thread-38", and "Client" seem to point at the jaco.mp3.player used to play the sound files for weather not stopping properly. Lines up with the experienced behavior.

So it looks like there's some interaction with that player and my system that causes the issue. I looked around a little bit and didn't see any similar postings so it's probably still highly specific to me.
